packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -0,0 +1,211 @@
+  }) : assert(tabBar != null),
+       assert(tabBuilder != null),
+       super(key: key);
+
+  /// The [tabBar] is a [CupertinoTabBar] drawn at the bottom of the screen
+  /// that lets the user switch between different tabs in the main content area
+  /// when present.
+  ///
+  /// When provided, [CupertinoTabBar.currentIndex] will be ignored and will
+  /// be managed by the [CupertinoTabScaffold] to show the currently selected page
+  /// as the active item index. If [CupertinoTabBar.onTap] is provided, it will
+  /// still be called. [CupertinoTabScaffold] automatically also listen to the
+  /// [CupertinoTabBar]'s `onTap` to change the [CupertinoTabBar]'s `currentIndex`
+  /// and change the actively displayed tab in [CupertinoTabScaffold]'s own
+  /// main content area.
+  ///
+  /// If translucent, the main content may slide behind it.
+  /// Otherwise, the main content's bottom margin will be offset by its height.
+  final CupertinoTabBar tabBar;
+
+  /// An [IndexedWidgetBuilder] that's called when tabs become active.
+  ///
+  /// The widgets built by [IndexedWidgetBuilder] is typically a [CupertinoTab]
+  /// in order to achieve the parallel hierarchies information architecture seen
+  /// on iOS apps with tab bars.
+  ///
+  /// When the tab becomes inactive, its content is still cached in the widget
+  /// tree [Offstage] and its animations disabled.
+  ///
+  /// Content can slide under the [tabBar] when it's translucent.
+  final IndexedWidgetBuilder tabBuilder;
+
+  @override
+  _CupertinoTabScaffoldState createState() => new _CupertinoTabScaffoldState();
+}
+
+class _CupertinoTabScaffoldState extends State<CupertinoTabScaffold> {
+  int _currentPage = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> stacked = <Widget>[];
+
+    // The main content being at the bottom is added to the stack first.
+    stacked.add(
+      new Padding(
+        padding: new EdgeInsets.only(bottom: widget.tabBar.opaque ? widget.tabBar.preferredSize.height : 0.0),
+        child: new _TabView(
+          currentTabIndex: _currentPage,
+          tabNumber: widget.tabBar.items.length,
+          tabBuilder: widget.tabBuilder,
+        )
+      ),
+    );
+
+    if (widget.tabBar != null) {
+      stacked.add(new Align(
+        alignment: FractionalOffset.bottomCenter,
+        // Override the tab bar's currentIndex to the current tab and hook in
+        // our own listener to update the _currentPage on top of a possibly user
+        // provided callback.
+        child: widget.tabBar.copyWith(
+            currentIndex: _currentPage,
+            onTap: (int newIndex) {
+              setState(() {
+                _currentPage = newIndex;
+              });
+              // Chain the user's original callback.
+              if (widget.tabBar.onTap != null)
+                widget.tabBar.onTap(newIndex);
+            }
+        ),
+      ));
+    }
+
+    return new Stack(
+      children: stacked,
```

Comment:
consider if you need to specify stackFit here, too

Reply:
thanks. Expand makes sense here